### PR TITLE
pidfds: switch infrastructure to rely on pidfds whenever possible

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: doc/api-extensions.md [hunk 1]
```diff
@@ -46,6 +46,8 @@ the kernel will generate a poll event and send a message over the file descripto
 
 The caller can read this message, inspect the syscalls including its arguments. Based on this information the caller is expected to send back a message informing the kernel which action to take. Until that message is sent the kernel will block the calling process. The format of the messages to read and sent is documented in seccomp itself.
 
+A new API function `seccomp_notify_fd()` has been which allows callers to retrieve the notifier fd for the container's seccomp filter.
```
Comment (stgraber) [inline doc/api-extensions.md]: `has been added`?

Comment (brauner) [reply doc/api-extensions.md]: oh yeah

**Comment by stgraber** [comment]
What's the upgrade handling for the pidfd logic. That is, we're running on a kernel which supports it but are running existing containers which were started with a liblxc version which doesn't support it.

Do we properly fallback to the old logic in that case?

**Comment by brauner** [comment]
> What's the upgrade handling for the pidfd logic. That is, we're running on a kernel which supports it but are running existing containers which were started with a liblxc version which doesn't support it.
> 
> Do we properly fallback to the old logic in that case?

Yeah, the callback required won't be available so we get a -1 and fallback to pids.

**Pull request merged**